program/src/instruction.rs

```diff
@@ -109,7 +114,7 @@ impl LidoInstruction {
         // `Borsh::serialize`, which takes an arbitrary writer, and which can
         // therefore return an IoError. But when serializing to a vec, there
         // is no IO, so for this particular writer, it should never fail.
-        self.try_to_vec()
+        self.try_to_vec() // actually it can fail on OutOfMemory error
```

Comment:
Out of memory should be unrecoverable, so that is fine.

(For very specialized use cases, a long-running daemon that is essential to the system might want to handle memory exhaustion, e.g. to do an emergency shutdown and persist some state to disk. But for programs running on a blockchain, it makes no sense to handle OOM; if you can edit the program to handle OOM, you might as well edit the program to make it more resource-efficient in the first place.)